module/core/src/Application/Request/ParamConverter/AggregateRootParamConverter.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Core\Application\Request\ParamConverter;
+
+use Ergonode\EventSourcing\Domain\AbstractAggregateRoot;
+use Ergonode\EventSourcing\Infrastructure\Manager\EventStoreManager;
+use Ergonode\SharedKernel\Domain\AggregateId;
+use Ergonode\Workflow\Domain\Entity\AbstractWorkflow;
+use Ergonode\Workflow\Domain\Provider\WorkflowProvider;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Webmozart\Assert\Assert;
+
+/**
+ */
+class AggregateRootParamConverter implements ParamConverterInterface
+{
+    /**
+     * @var EventStoreManager
+     */
+    private EventStoreManager $manager;
+
+    /**
+     * @param EventStoreManager $manager
+     */
+    public function __construct(EventStoreManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function apply(Request $request, ParamConverter $configuration): void
+    {
+        $name = $configuration->getName();
+        Assert::notNull($name);
+
+        $parameter = $request->get($name);
+        Assert::notNull($parameter);
+
+        if (!AggregateId::isValid($parameter)) {
```

Comment:
Seems like `$parameter` can still not be string I think - please verify

Reply:
But method 'isValid' checking if it uuid.

In AggregateRootParamConvterTest thers is even test for this:

```
 /**
     */
    public function testInvalidParameter(): void
    {
        $this->expectException(BadRequestHttpException::class);
        $this->configuration->method('getName')->willReturn('test');
        $this->configuration->method('getClass')->willReturn(Category::class);
        $this->request->method('get')->willReturn('test');

        $paramConverter = new AggregateRootParamConverter($this->manager);
        $paramConverter->apply($this->request, $this->configuration);
    }
```

Reply:
Id does but it expects string as argument. You're checking whether it's null. So potentially it might i.e. integer as well.